components/social/org.wso2.carbon.social.core/src/main/java/org/wso2/carbon/social/core/service/SocialActivityService.java

```diff
@@ -71,6 +71,17 @@ public long publish(String activity) throws SocialActivityException {
 	}
 
 	/**
+	 * Update an activity given new activity
+	 *
+	 * @param activity
+	 * @throws SocialActivityException
+	 * @throws JsonSyntaxException
+	 */
+	public String update(String activity) throws SocialActivityException {
+		return getActivityPublisher().update(activity);
+	}
+
```

Comment:
Thanks @daneshk   , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/fbeb6d47ba62123013110cde735ae787af024629)